onnx/test/shape_inference_test.py

```diff
@@ -2111,7 +2125,7 @@ def test_if_with_different_optional_shapes_in_then_else_branches(self) -> None:
             []
         )
 
-        output_tensor_proto = helper.make_tensor_type_proto(elem_type=TensorProto.FLOAT, shape=None)
+        output_tensor_proto = helper.make_tensor_type_proto(elem_type=TensorProto.FLOAT, shape=())
```

Comment:
I don't think `shape=()` in this example. The then branch produces a `shape=[1]` and else branch produces a `shape=[5]`.

Reply:
I think we should use `shape=(None,)` or `shape=[None]` here

Reply:
Updated since (None, ) is more precise as you suggested. Thank you @gramalingam !